maia-hdl/projects/pluto/system_bd.tcl

```diff
@@ -5,6 +5,9 @@
 # We need to do this here because adi_project_create overwrites whatever we had
 # set beforehand.
 set_property ip_repo_paths {../../ip ../../adi-hdl/library} [current_fileset]
+if {[info exists maia_iio]} {
+add_files -norecurse  ../../adi-hdl/library/common/ad_bus_mux.v
+}
```

Comment:
This should go in `system_project.tcl`. The `system_bd.tcl` file is about generating the block design. Adding files to the project is done in `system_project.tcl`.

Reply:
Also it looks like this file is already added to the project in `pluto_iio` but not in `plutoplus_iio`.

Reply:
Removing this line in system_bd.tcl and results in an error as the component is used here : https://github.com/F5OEO/maia-sdr/blob/5fc18c2da9d61cce4beb64f2930f2cc913edf7a3/maia-hdl/projects/pluto/system_bd.tcl#L428 . Even it is included in system_project.tcl. Let me know if you have an other way to reference the component without include it on system_bd.tcl

Reply:
You're right. I've looked into this and the problem is that `adi_project_create` is the one that runs `source system_bd.tcl`. This is before `adi_project_files` has run, so `system_bd.tcl` doesn't have any of the files added in `adi_project_files` available.

Therefore, my suggestion is to move the line `add_files -norecurse  ../../adi-hdl/library/common/ad_bus_mux.v` below, immediately before `ad_bus_mux` is instantiated in the block design, and to remove `"$ad_hdl_dir/library/common/ad_bus_mux.v"` from `adi_project_files`, since the file will have already been added by `system_bd.tcl` when running `adi_project_create`.